### PR TITLE
Allow TabItem to be Disabled

### PR DESCRIPTION
**Pull request by gavinwade12**
### Description:

Addresses #1904 by adding methods to AppTabs and DocTabs for disabling TabItems.

### Checklist:
- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
~~I was having the issue with the dialog tests failing, so I installed Go 1.17 (currently on 1.18.5) and ran the tests with it. However, I'm still having issues with (seemingly) unrelated tests failing that are also failing in other PRs. Here's an [example](https://github.com/fyne-io/fyne/runs/7603055303?check_suite_focus=true); although, it's strange the 1.14-ubuntu failed and 1.17-ubuntu passed here when I am running the tests with 1.17 on Linux Mint.~~
I have an open PR for the randomly-failing test (#3175).

#### Where applicable:
- [x] Public APIs match existing style and have Since: line.

### CODE REVIEW

File: container/apptabs.go [hunk 1]
```diff
@@ -89,6 +88,34 @@ func (t *AppTabs) CurrentTabIndex() int {
 	return t.current
 }
 
+// Enable enables the specified TabItem.
+//
+// Since: 2.2
```
Comment (andydotxyz) [inline container/apptabs.go:93]: 2.2 is already released, this would need to be 2.3

Comment (andydotxyz) [reply container/apptabs.go:93]: Also we have a `Disablable` interface with `Enable` which enables the whole widget, so this name overloading may be confusing. I would recommend `EnableItem` to indicate that it's function is different (same with `Disable` etc).

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Thanks for the contribution.

A few thoughts inline.
Plus you'll need to address the static analysis failures

File: container/apptabs.go (hunk 1 above)
Comment (gavinwade12) [reply container/apptabs.go]: Thanks for the review! I believe I've made your suggested updates, and the static analysis is now passing. There is now just the `TestForm_Disabled` test that is sometimes failing, but I believe it's unrelated to this change. The same test fails when I run it prior to my changes e.g. `git checkout 2504bea3c6a5c97a1d5b45d7c3a39128de6fba5a && go test ./...`.

Comment (andydotxyz) [reply container/apptabs.go]: There does seem to be an instability in that form test. All others are now fixed but that one remains a mystery

Comment (gavinwade12) [reply container/apptabs.go]: I believe I may have solved the issue with the form test: https://github.com/fyne-io/fyne/pull/3175.

File: container/tabs.go [hunk 2]
```diff
@@ -29,22 +29,22 @@ func (ti *TabItem) enable() {
 	}
 }
 
+func (ti *TabItem) disable() {
```
Comment (andydotxyz) [inline container/tabs.go:32]: private functions (not exported) should be below public (exported) ones.
Apologies that this is not automatically enforced yet, we are working on it.
More info about order of code in a file at: https://github.com/fyne-io/fyne/wiki/Contributing#code-style

File: container/apptabs.go [hunk 3]
```diff
@@ -89,6 +88,34 @@ func (t *AppTabs) CurrentTabIndex() int {
 	return t.current
 }
 
+// EnableItem enables the specified TabItem.
```
Comment (andydotxyz) [inline container/apptabs.go:91]: We try to keep stuff alphabetical, please put enable after disable.
https://github.com/fyne-io/fyne/wiki/Contributing#code-style

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Looking good, but can you change the order of functions please - noted the contributor doc inline.
We really need to make that an automated check...

File: container/tabs.go (hunk 2 above)
Comment (gavinwade12) [reply container/tabs.go:36]: Sorry about that! I admit I should have read through that a bit more thoroughly the first time. I believe things are correct now.

File: container/apptabs.go (hunk 3 above)
Comment (gavinwade12) [reply container/apptabs.go]: This is fixed now. My apologies for the delay!

**Comment by gavinwade12** [comment]
Hey Andy, my apologies for the delay in cleaning this up. I think it's ready to go now!

**Review by andydotxyz — APPROVED** [review]
Great, thanks!

**Pull request merged**